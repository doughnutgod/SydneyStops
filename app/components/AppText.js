import React from "react";

import { Text, StyleSheet, Platform } from "react-native";
//import AppColors from '../config/AppColors';

//Handles a template for text
function AppText({ children, style }) {
  return <Text style={[styles.text, style]}> {children} </Text>;
}

const styles = StyleSheet.create({
  text: {
    fontSize: 14,
    fontFamily: Platform.OS === "android" ? "sans-serif-medium" : "Arial",
  },
});

export default AppText;
