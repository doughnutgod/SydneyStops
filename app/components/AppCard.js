import React from 'react';
import { StyleSheet, View, Image, Text} from 'react-native';

import AppColors from '../config/AppColors';
import AppText from '../components/AppText';
import AppFont from '../config/AppFont';


//Handles a template for Cards
function AppCard({name, city, image, date, category, user}) {
    return (
        <View style={styles.container}>
            {isFinite(image) ? <Image source={image} style={styles.image}/> : <><Image source={{uri: image}} style={styles.image}/></>}
            <AppText style={styles.title}>Establishment Name: {name}</AppText>
            {city && <AppText style={styles.subtitle}>City: {city}</AppText>}
            {date && <AppText style={styles.subtitle}>Date Visited: {date}</AppText>}
            {category && <AppText style={styles.subtitle}>Category: {category}</AppText>}
            {user && <AppText style={styles.subtitle}>Submitted By: {user}</AppText>}
        </View>
    );
}

const styles = StyleSheet.create({
    container:{
        backgroundColor:AppColors.AlternativeColor,
        borderRadius:20,
        overflow: 'hidden',
        padding: 10,
        marginBottom: 20
    },
    image:{
        borderRadius:15,
        height:220,
        width:"100%"
    },
    title:{
        fontSize: AppFont.CardInfoName
    },
    subtitle:{
        fontSize: AppFont.CardInfoOther
    },
})
export default AppCard;