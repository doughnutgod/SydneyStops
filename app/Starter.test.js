import React from "react";
import renderer from 'react-test-renderer';
import AppText from './components/AppText';
import AppCard from './components/AppCard';


test('AppText will render', () => {
    const json = renderer.create(<AppText />).toJSON();
    expect(json.props.style[0].fontSize).toBe(14);
});

test('AppText will render', () => {
    const json = renderer.create(<AppCard />).toJSON();
    expect(json.props.style.borderRadius).toBe(20);
});

test('AppText renders correctly', () => {
    const tree = renderer.create(<AppCard />).toJSON();
    expect(tree).toMatchSnapshot();
});