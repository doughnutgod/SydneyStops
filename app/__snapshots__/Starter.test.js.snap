// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppText renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#666666",
      "borderRadius": 20,
      "marginBottom": 20,
      "overflow": "hidden",
      "padding": 10,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 15,
        "height": 220,
        "width": "100%",
      }
    }
  />
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Arial",
          "fontSize": 14,
        },
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
     
    Establishment Name: 
     
  </Text>
</View>
`;
